overlord/snapstate/snapmgr.go

```diff
@@ -830,20 +835,7 @@ func Manager(st *state.State, runner *state.TaskRunner) (*SnapManager, error) {
 	// We cannot undo much after a component file is removed. And it is the
 	// last task anyway.
 	runner.AddHandler("discard-component", m.doDiscardComponent, nil)
-	setupKModsInDo := func(t *state.Task, _ *tomb.Tomb) error {
-		return m.doSetupKernelModules(t, state.DoneStatus)
-	}
-	setupKModsInUndo := func(t *state.Task, _ *tomb.Tomb) error {
-		return m.doSetupKernelModules(t, state.UndoneStatus)
-	}
-	removeKModsInUndo := func(t *state.Task, _ *tomb.Tomb) error {
-		return m.doRemoveKernelModulesSetup(t, state.UndoneStatus)
-	}
-	removeKModsInDo := func(t *state.Task, _ *tomb.Tomb) error {
-		return m.doRemoveKernelModulesSetup(t, state.DoneStatus)
-	}
-	runner.AddHandler("prepare-kernel-modules-components", setupKModsInDo, removeKModsInUndo)
-	runner.AddHandler("clear-kernel-modules-components", removeKModsInDo, setupKModsInUndo)
+	runner.AddHandler("prepare-kernel-modules-components", m.doSetupKernelModulesMany, m.undoSetupKernelModulesMany)
```

Comment:
it's hard to tell just from the diff, those this has proper do-undo tests at the level of handler testing?

Reply:
Yes, `TestSetupKernelModules`, `TestSetupKernelModulesFails`, `TestRemoveKernelModulesSetup`, and `TestRemoveKernelModulesSetupFails` test the do/undo logic.

Reply:
I see now we have a bunch of tasks related to kernels, where the implementation methods and the task name don't match, that's confusing. Also maybe we should do a quick pass to make sure we are happy with the task names?

Reply:
The tasks names (`prepare-kernel-snap`, `prepare-kernel-modules-components`) seem coherent, maybe we can change `doSetupKernelSnap`/`doSetupKernelModulesMany` to match.

Reply:
I renamed them in 9523f18faeb7ca2acd4fdc636fb7eddff6099101 and 0937bb2ebb89551456fca69cd3b69b48aaad9592.

Reply:
Thanks! But somewhat they do not look they are around, both show the "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository." when I follow the link.

Reply:
I think that is just what github shows when commits are in a fork of the destination repo/PR. You can see the last commit added to this branch is one of the commits there.

Reply:
Ok, I see the changes in the diff. I wonder if it would help to do rebases instead or merging master which at least to me it makes history more difficult read.

Reply:
I don't totally disagree, but I think rebasing can mess with github's functionality to show changes from a last review. Additionally, we have in CONTRIBUTING.md to not force-push a branch once it has received reviews.